mlflow/utils/credentials.py

```diff
@@ -76,29 +77,25 @@ def login(backend="databricks"):
         )
 
 
-def _check_databricks_auth():
-    # Check if databricks credentials are set.
-    try:
-        from databricks.sdk import WorkspaceClient
-    except ImportError:
-        raise ImportError(
-            "Databricks SDK is not installed. To use `mlflow.login()`, please install "
-            "databricks-sdk by `pip install databricks-sdk`."
-        )
+def _validate_databricks_auth():
+    """Validate Databricks authentication."""
+    max_trials = 3  # Allow 3 retrys for timeout.
+    timeout = 3
+    for _ in range(max_trials):
+        try:
+            result = subprocess.run(
+                ["databricks", "tokens", "list"],
```

Comment:
can we validate `databricks` is available? If not, we can throw immediately. No retry is required.

Reply:
Any ohter scenarios where retry doesn't make sense?

Reply:
I think `databricks-cli` is bundled with `mlflow`? I think we can assume it's existing inside MLflow repo?

Reply:
Yes, mlflow depends on `databricks-cli`. I think we'll remove `databricks-cli` in the future and migrate to the new CLI (which is a CLI written in Go).